akka-stream-tests/src/test/scala/akka/stream/scaladsl/SetupSpec.scala

```diff
@@ -193,7 +193,7 @@ class SetupSpec extends StreamSpec {
         }
         .named("my-name")
 
-      Source.empty.runWith(sink).flatten.futureValue shouldBe Some("setup-my-name")
+      Source.empty.runWith(sink).flatten.futureValue shouldBe Some("my-name-setup")
```

Comment:
Frustratingly, the new implementation caused this to flip again (due to an additional "layer" in the graph). Admittedly it's not perfect ("setup-my-name" does read more easily) but I just don't have the bandwidth to fight this any further given that it essentially only affects the `nameLifted` attribute (which accumulates instead of overrides). We have a more critical test for attribute order that continues to pass: `give priority to attributes of inner sink`.

Reply:
I think this is more correct though, in line with how attributes in general work, since "my-name" is less "specific"  than the "setup" name.

Reply:
Well, this is correct in that it matches how the stage is implemented. However, from a user perspective I would expect that explicitly naming a stage will add my name to the end (i.e., most specific). So arguably we are leaking implementation details here.